packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -96,5 +110,35 @@ export class Communicator {
         if (!this.popup) throw standardErrors.rpc.internal();
         return this.popup;
       });
-  }
+  };
+
+  /**
+   * Posts a RPC request to the popup window and waits for a response
+   */
+  postRPCRequest = (request: RPCRequestMessage): Promise<RPCResponseMessage> => {
+    this.pendingRequests.set(request.id, 'created');
```

Comment:
dapp "creates" a request that hasn't been awaited yet.
e.g. `const promise = communicator.postRequest(request)`